third_party/move/move-vm/types/src/resolver.rs

```diff
@@ -30,4 +30,15 @@ pub trait ResourceResolver {
         metadata: &[Metadata],
         layout: Option<&MoveTypeLayout>,
     ) -> PartialVMResult<(Option<Bytes>, usize)>;
+
+    fn get_resource_size_with_metadata_and_layout(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+        metadata: &[Metadata],
+        layout: Option<&MoveTypeLayout>,
+    ) -> PartialVMResult<Option<u64>> {
+        let (bytes, bytes_loaded) = self.get_resource_bytes_with_metadata_and_layout(address, struct_tag, metadata, layout)?;
+        Ok(bytes.map(|_| bytes_loaded as u64))
```

Comment:
Because of resource groups, `bytes.len() != bytes_loaded` sometimes. Let's add documentation that this returns "total loaded size which may not be equal to length of bytes" or something like that?